core/platform/android/java/src/dev/axmol/lib/AxmolRenderer.java

```diff
@@ -47,9 +47,9 @@ public class AxmolRenderer implements GLSurfaceView.Renderer {
     private long mLastTickInNanoSeconds;
     private int mScreenWidth;
     private int mScreenHeight;
-
     private static boolean gNativeInitialized = false;
     private static boolean gNativeIsPaused = false;
+    private boolean mNativeInitCompleted = false;
```

Comment:
the variable `gNativeInitialized` already indicate whether native inited.

Reply:
@halx99 
yeah `gNativeInitialized` is static so at first launch the AppActivity will be created along with the SurfaceView and the Renderer and the value will be true afterward no matter what. 
Now when a Warm start occurs, a new AppActivity will be instanciated, creating a new SurfaceView and of course a new Renderer, which means the static `gNativeInitialized` 
 will still be true but `mNativeInitCompleted` will be false as it's a new instance, that's how we figure out a Warm start as occured.